### PR TITLE
tests: bail on status leadership mismatch

### PR DESCRIPTION
**Pull request by acpana**
This change builds on #5886 in order to make testing easier!

### CODE REVIEW

File: pkg/controller/kccmanager/kccmanager_multicluster_test.go
Comment (gemmahou) [inline pkg/controller/kccmanager/kccmanager_multicluster_test.go:1]: /lgtm

nit: Do we need a test case for the scenario where GlobalHolderIdentity is nil? (e.g. the MultiClusterLease object fails to reconcile, and has empty status)

Comment (acpana) [reply pkg/controller/kccmanager/kccmanager_multicluster_test.go:1]: done!

File: pkg/controller/kccmanager/kccmanager.go [hunk 1]
```diff
@@ -139,7 +140,7 @@ func setUpMultiClusterLease(ctx context.Context, restConfig *rest.Config, scheme
 		if err := c.Get(ctx, ccName, cc); err != nil {
 			// If the ConfigConnector object is not found, proceed with default leader election.
 			klog.Infof("ConfigConnector object not found, using default in-cluster leader election")
```
Comment (maqiuyujoyce) [inline pkg/controller/kccmanager/kccmanager.go:142]: Not a blocker: I'd log the actual error here in case it is an edge case.

File: pkg/controller/kccmanager/kccmanager.go [hunk 2]
```diff
@@ -176,22 +177,51 @@ func setUpMultiClusterLease(ctx context.Context, restConfig *rest.Config, scheme
...
 
 func (m *leaderElectionManager) Start(ctx context.Context) error {
 	m.leConfig.Callbacks.OnStartedLeading = func(ctx context.Context) {
 		klog.Infof("started leading; identity: %s", m.leConfig.Lock.Identity())
+
+		// Explicitly fetch the MultiClusterLease object to verify its status.
+		lease := &mclv1alpha1.MultiClusterLease{}
+		leaseName := types.NamespacedName{Namespace: m.mclConfig.Namespace, Name: m.mclConfig.LeaseName}
+		if err := m.Manager.GetAPIReader().Get(ctx, leaseName, lease); err != nil {
+			m.fatal("error fetching MultiClusterLease %s: %v", leaseName, err)
```
Comment (maqiuyujoyce) [inline pkg/controller/kccmanager/kccmanager.go:212]: Agent mentioned the following and I'm curious too as it looks that we'll implement onFatal() separately.
> Does `m.fatal` ensure that the process exits with a non-zero status code?
In the default implementation it calls `klog.Fatalf`, which does.
Just verifying that this aligns with the "bail" requirement.

File: pkg/controller/kccmanager/kccmanager.go [hunk 3]
```diff
@@ -176,22 +177,51 @@ func setUpMultiClusterLease(ctx context.Context, restConfig *rest.Config, scheme
...
+	klog.Fatalf(format, args...)
 }
 
 func (m *leaderElectionManager) Start(ctx context.Context) error {
 	m.leConfig.Callbacks.OnStartedLeading = func(ctx context.Context) {
 		klog.Infof("started leading; identity: %s", m.leConfig.Lock.Identity())
+
+		// Explicitly fetch the MultiClusterLease object to verify its status.
+		lease := &mclv1alpha1.MultiClusterLease{}
+		leaseName := types.NamespacedName{Namespace: m.mclConfig.Namespace, Name: m.mclConfig.LeaseName}
```
Comment (maqiuyujoyce) [inline pkg/controller/kccmanager/kccmanager.go:210]: Agent mentioned the following. I'm not super opinionated on it.

> `m.mclConfig` is assumed to be non-nil here. While `New` ensures it is populated when `leConfig` is present, adding a sanity check or comment in `Start` or `New` might be safer against future refactors.
